src/ds3.c

```diff
@@ -790,6 +790,22 @@ void ds3_request_set_max_keys(ds3_request* _request, uint32_t max_keys) {
     _set_query_param(_request, "max-keys", max_keys_s);
 }
 
+void ds3_request_set_name(ds3_request* _request, const char* name) {
+    _set_query_param(_request, "name", name);
+}
+
+void ds3_request_set_id(ds3_request* _request, const char* id) {
+    _set_query_param(_request, "id", id);
+}
+
+void ds3_request_set_type(ds3_request* _request, const char* type) {
```

Comment:
Is there a finite set of types?  If so we should probably create an enum and use that instead of a string.